src/cert_context.rs

```diff
@@ -488,6 +565,16 @@ impl KeySpec {
     }
 }
 
+/// Valid uses of a Certificate - All, None, or specific OIDs
+pub enum ValidUses {
+    /// Certificate is valid for all uses
+    All,
+
+    /// Certificate is valid for uses specified. No entries means that the certificate
+    /// has no valid uses.
+    OIDs(Vec<String>),
```

Comment:
This should be called `Oids` I believe, to match up with abbreviation capitalization conventions.